src/client/App.tsx

```diff
@@ -217,7 +217,18 @@ function App() {
   useEffect(() => {
     // Sync comments whenever they change
     if (comments.length > 0) {
-      const data = JSON.stringify({ comments });
+      // Transform DiffComment to Comment format for server
+      const transformedComments = comments.map((c) => ({
+        id: c.id,
+        file: c.filePath,
+        line:
+          typeof c.position.line === 'number' ?
+            c.position.line
+          : [c.position.line.start, c.position.line.end],
+        body: c.body,
+        timestamp: c.createdAt,
+      }));
+      const data = JSON.stringify({ comments: transformedComments });
```

Comment:
The logic is duplicated, but with subsequent modifications, it will no longer be necessary to convert the DiffComment type into the Comment type in the first place.